### ISSUE TITLE
quotes "avoid-escape" option doesn't work if you use backticks to avoid escape

### ISSUE THREAD

**Issue by callumlocke** [eslint/eslint#5611]
**What version of ESLint are you using?**

v2.4.0

**What parser are you using?**

babel-eslint

**Please show your full configuration:**

```yaml
parser: babel-eslint
rules:
  quotes: [2, single, avoid-escape]
```

**What did you do? Please include the actual source code causing the issue.**

```js
const foo = `I'm using backticks to avoid having to 'escape' the single-quotes inside the string`;
```

**What did you expect to happen?**

For eslint not to complain, because I've used the "avoid-escape" option.

**What actually happened?**

eslint said it was an error:

```
~/demo.js
  1:11  error  Strings must use singlequote  quotes
```

**Comment by codekirei** [comment]
Disclaimer &mdash; I'm not a maintainer, but it looks like this issue stems from the `quotes` rule hard-coding the `alternateQuote` char [here](https://github.com/eslint/eslint/blob/master/lib/rules/quotes.js#L20):

`"` -> `'`
`'` -> `"`
`` ` `` -> `"`

This could be potentially be avoided by allowing either of the other two quote chars as an alternate in the `isValid` test [here](https://github.com/eslint/eslint/blob/master/lib/rules/quotes.js#L178).

In other words, this doesn't appear difficult to fix technically, so it's more of a design decision on whether it makes sense to allow either char as an alternate for escape strings. That's not up to me :smile:

**Comment by platinumazure** [comment]
I could be convinced that this is a bug, but I don't know if users are "expected" to use a different rule configuration when they want to use backticks for escaping.

@eslint/eslint-team Can anyone speak to the intended design here, so we can know if this is a bug or an enhancement request?

**Comment by alberto** [comment]
It's not a bug. From the rule doc:

> When using `"single"` or `"double"`, template literals that don't contain a substitution, don't contain a line break and aren't tagged templates, are flagged as problems, even with the `"avoid-escape"` option.

**Comment by alberto** [comment]
This is a duplicate of https://github.com/eslint/eslint/issues/5234
